novawallet/Modules/Staking/NominationPools/BondMore/Setup/NominationPoolBondMoreSetupPresenter.swift

```diff
@@ -0,0 +1,177 @@
+        baseView as? NominationPoolBondMoreSetupViewProtocol
+    }
+
+    var wireframe: NominationPoolBondMoreSetupWireframeProtocol? {
+        baseWireframe as? NominationPoolBondMoreSetupWireframeProtocol
+    }
+
+    var interactor: NominationPoolBondMoreSetupInteractorInputProtocol? {
+        baseInteractor as? NominationPoolBondMoreSetupInteractorInputProtocol
+    }
+
+    private var inputResult: AmountInputResult?
+
+    init(
+        interactor: NominationPoolBondMoreSetupInteractorInputProtocol,
+        wireframe: NominationPoolBondMoreSetupWireframeProtocol,
+        chainAsset: ChainAsset,
+        hintsViewModelFactory: NominationPoolsBondMoreHintsFactoryProtocol,
+        balanceViewModelFactory: BalanceViewModelFactoryProtocol,
+        dataValidatorFactory: NominationPoolDataValidatorFactoryProtocol,
+        localizationManager: LocalizationManagerProtocol,
+        logger: LoggerProtocol
+
+    ) {
+        super.init(
+            interactor: interactor,
+            wireframe: wireframe,
+            chainAsset: chainAsset,
+            hintsViewModelFactory: hintsViewModelFactory,
+            balanceViewModelFactory: balanceViewModelFactory,
+            dataValidatorFactory: dataValidatorFactory,
+            localizationManager: localizationManager,
+            logger: logger
+        )
+    }
+
+    func provideTransferrableBalance() {
+        guard let balance = assetBalance?.transferable.decimal(precision: chainAsset.asset.precision) else {
+            view?.didReceiveTransferable(viewModel: nil)
+            return
+        }
+        let viewModel = balanceViewModelFactory.amountFromValue(balance).value(for: selectedLocale)
+        view?.didReceiveTransferable(viewModel: viewModel)
+    }
+
+    func provideAmountInputViewModel() {
+        let inputAmount = getInputAmount()
+        let viewModel = balanceViewModelFactory.createBalanceInputViewModel(
+            inputAmount
+        ).value(for: selectedLocale)
+
+        view?.didReceiveInput(viewModel: viewModel)
+    }
+
+    func provideAssetViewModel() {
+        let balance = assetBalance?.transferable.decimal(precision: chainAsset.asset.precision) ?? 0
+        let inputAmount = getInputAmount() ?? 0
+        let viewModel = balanceViewModelFactory.createAssetBalanceViewModel(
+            inputAmount,
+            balance: balance,
+            priceData: price
+        ).value(for: selectedLocale)
+
+        view?.didReceiveAssetBalance(viewModel: viewModel)
+    }
+
+    override func getInputAmount() -> Decimal? {
+        guard let inputResult = inputResult else {
+            return nil
+        }
+        switch inputResult {
+        case .rate:
+            let fee = fee?.decimal(precision: chainAsset.asset.precision) ?? 0
+            let availableAmountDecimal = assetBalance?.transferable.decimal(precision: chainAsset.asset.precision) ?? 0
+            return inputResult.absoluteValue(from: availableAmountDecimal - fee)
```

Comment:
we should take ed into account here. So, let have min(transferable - fee, total - fee - ed)